crates/testing/tests/consensus_task.rs

```diff
@@ -10,13 +10,283 @@ use hotshot_types::{
 use jf_primitives::vid::VidScheme;
 use std::collections::HashMap;
 
+fn permute<T>(inputs: Vec<T>, order: Vec<usize>) -> Vec<T>
```

Comment:
This function is going to be deleted once [this](https://github.com/EspressoSystems/HotShot/pull/2697) merges. The branches diverged due to an in-between PR with this one, so I'll move this out to a central location once that goes in.